python/tests/io/test_pyarrow.py

```diff
@@ -1527,3 +1527,19 @@ def test_writing_avro_file_gcs(generated_manifest_entry_file: str, pyarrow_filei
             assert b1 == b2  # Check that bytes of read from local avro file match bytes written to s3
 
     pyarrow_fileio_gcs.delete(f"gs://warehouse/{filename}")
+
+
+def test_parse_hdfs_location() -> None:
```

Comment:
Ah, I see. The other arrow file systems require the scheme to be removed, but for hdfs you can access both local files and on hdfs.